packages/booter-lb3app/src/lb3app.booter.ts

```diff
@@ -5,6 +5,7 @@
 
 import {BootBindings, Booter} from '@loopback/boot';
 import {CoreBindings, inject} from '@loopback/core';
+import {DataSource} from '@loopback/repository';
```

Comment:
As I commented before, `@loopback/repository` is not in regular/peer dependencies of `booter-lb3app`, therefore it may not be available at runtime, when this booter is installed in a LB project.

Also the `DataSource` type is not the right one to use, it described LB4 objects but your code is using LB3 instances! The correct type is `juggler.DataSource`.

Please remove this import and find a different solution how to describe the shape (interface) of LB3 datasource objects.